data_pipeline/generic_web_api/generic_web_api_data_etl.py

```diff
@@ -157,18 +160,27 @@ def generic_web_api_data_etl(
             )
             items_count = len(items_list)
 
+            from_date_to_advance, to_reset_other_request_param = (
```

Comment:
It would be good to have tests that cover this functionality.

I found the naming a bit confusing at first. Maybe `should_reset_pagination` would be slightly clearer. That should be true whenever the start date has changed. Maybe moving that logic out of `get_next_start_date` could make it a bit more obvious.